cpp-data-aggregation/aggregation_consumer.cpp

```diff
@@ -0,0 +1,141 @@
+#include "client.h"
+#include <mpi.h>
+
+
+int get_iterations() {
+  char* iterations = std::getenv("SS_ITERATIONS");
+  int iters = iterations ? std::stoi(iterations) : 100;
+  return iters;
+}
+
+void run_aggregation_consumer(std::ofstream& timing_file,
+                              int list_length)
+{
+    int rank;
+    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+
+    if (rank == 0)
+        std::cout << "Connecting clients" << std::endl;
+
+    // Connect the client and save connection time
+    double constructor_start = MPI_Wtime();
+    SmartRedis::Client client(true);
+    double constructor_end = MPI_Wtime();
+    double delta_t = constructor_end - constructor_start;
+    timing_file << rank << "," << "client()" << ","
+                << delta_t << "\n";
+
+    // Allocate arrays to hold timings
+    std::vector<double> get_list_times;
+
+    // Retrieve the number of iterations to run
+    int iterations = get_iterations();
+
+    // Block to make sure all clients are connected
+    MPI_Barrier(MPI_COMM_WORLD);
+
+    // Retrieve rank-local loop start time
+    double loop_start = MPI_Wtime();
+
+    // Perform dataset aggregation retrieval
+    for (int i=0; i<iterations; i++) {
+
```

Comment:
I'd suggest putting in an `MPI_Barrier` here just on the off chance that `client.delete_list` takes forever to return. At the very least we'll know that all ranks are perfectly synced at the start of the iterations

Reply:
Agreed on wanting to sync for each iteration.  I think the ``MPI_Barrier`` below where we check that the list is empty will synchronize all of the ranks before any new datasets are added to the aggregation list and datasets are pulled from the aggregation list.  I think the extra ``MPI_Barrier`` might just add extra time to the loop without more sync.

Reply:
Ah yeah ok, I see that. I agree no need for `MPI_Barrier`